app/components/TrackCard/index.js

```diff
@@ -0,0 +1,114 @@
+/**
+ *
+ * Track Card
+ *
+ */
+
+import React, { useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { Col, Button } from 'antd';
+import T from '@components/T';
+import If from '@components/If';
+import { isEmpty } from 'lodash';
+import { Link } from 'react-router-dom';
+
+const CustomCard = styled(Col)`
+  && {
+    margin: 20px 0;
+    ${(props) => props.span};
+    cursor: pointer;
+    border: 1px solid gray;
+    padding: 10px;
+  }
+`;
+
+const TrackImg = styled.img`
+  && {
+    text-align: center;
+  }
+`;
+
+export function TrackCard({
+  trackName,
+  collectionName,
+  previewUrl,
+  trackId,
+  artworkUrl100,
+  dispatchCurrentTrack,
+  currentTrack
+}) {
+  const audioRef = useRef(null);
+  const [isPlaying, setIsPlaying] = useState(false);
+  // const [currentSong, setCurrentSong] = useState(null);
+
+  useEffect(() => {
+    if (audioRef.current) {
+      if (currentTrack !== trackId) {
+        audioRef.current.pause();
+        setIsPlaying(false);
+      }
+    }
+  }, [currentTrack]);
+
+  const playHandler = () => {
+    // eslint-disable-next-line no-console
+    console.log(audioRef.current.paused);
+    if (audioRef.current.paused) {
+      dispatchCurrentTrack(trackId);
+      audioRef.current.play();
+      setIsPlaying(true);
+    } else {
+      dispatchCurrentTrack(null);
+      audioRef.current.pause();
+      setIsPlaying(false);
+    }
+    setIsPlaying(!isPlaying);
```

Comment:
we are already setting `isPlaying` in the if/else block, wouldn't this just revert the state change and set it to previous state

Reply:
Correct, I was doing that to toggle btn text, but that's not needed.